test/e2e/ipam/ipamhelper.go

```diff
@@ -0,0 +1,225 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package ipam is a helper to claim ip addresses from an IPAM provider cluster.
+package ipam
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/rand"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/utils/ptr"
+	ipamv1 "sigs.k8s.io/cluster-api/exp/ipam/api/v1alpha1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/yaml"
+
+	. "sigs.k8s.io/cluster-api-provider-vsphere/test/e2e/helper"
+)
+
+var ipamScheme *runtime.Scheme
+
+const controlPlaneEndpointVariable = "CONTROL_PLANE_ENDPOINT_IP"
+
+func init() {
+	ipamScheme = runtime.NewScheme()
+	_ = ipamv1.AddToScheme(ipamScheme)
+}
+
+// ConfigureControlPlaneIPs claims one or more ip addresses if the environment variable CONTROL_PLANE_ENDPOINT_IP is not set.
+// It returns a path to a new clusterctl configuration, which contains the variable for CONTROL_PLANE_ENDPOINT_IP as well as
+// for additional variables passed via additionalIPVariableNames and a cleanup function which is supposed to get called
+// in an AfterEach ginkgo function.
```

Comment:
This comment seems to be outdated (as we always get IPs when e2eIPAMKubeconfig is set)

Reply:
whoops, that was by accident Oo, don't know how I did that sorry

Reply:
I wanted to append it to my comment here 🤦  https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/pull/2612#discussion_r1452343106 wtf